src/OmniSharp.LanguageServerProtocol/Helpers.cs

```diff
@@ -96,7 +97,41 @@ public static DiagnosticSeverity ToDiagnosticSeverity(string logLevel)
         }
 
         public static DocumentUri ToUri(string fileName) => DocumentUri.File(fileName);
-        public static string FromUri(DocumentUri uri) => uri.GetFileSystemPath().Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar);
+
+        public static DocumentUri ToUri(MetadataSource mds) =>
+            new Uri($"osmd:/Project/{mds.ProjectName}/Assembly/{mds.AssemblyName}/Symbol/{mds.TypeName}.cs");
+
+        public static string FromUri(DocumentUri uri) => uri.Scheme switch {
+            "file" => uri.GetFileSystemPath().Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar),
+            "osmd" => FromOsmdUriPath(uri.Path),
```

Comment:
I would just go with `omnisharp` here to make it clear where it's coming from.

Reply:
Done.